gnus-recent.el

```diff
@@ -165,60 +216,282 @@ Warn if RECENT can't be deconstructed as expected."
+            (read
+             (with-temp-buffer
+               (insert-file-contents gnus-recent-file)
+               (buffer-string)))
+          nil))
+  (gnus-message 5 "Read %d item(s) from %s... done."
+                (length gnus-recent--articles-list) gnus-recent-file))
+
+(defun gnus-recent-count-saved ()
+  "Count the number of articles saved in `gnus-recent-file'."
+  (if (file-readable-p gnus-recent-file)
+      (length (read
+               (with-temp-buffer
+                 (insert-file-contents gnus-recent-file)
+                 (buffer-string))))
+    nil))
+
+;;
+;; Track new messages
+;;
+(defun gnus-recent--track-message ()
+  "Add an newly sent message to the list of tracked articles.
+Is run from `message-sent-hook'. A alist of the message header
+data should be available on `gnus-recent--temp-message-headers'."
+  (interactive)
+  (let* ((hdrs gnus-recent--temp-message-headers)
+         (date (gnus-recent-date-format (alist-get 'date hdrs)))
+         (author (rfc2047-decode-address-string (or (alist-get 'from hdrs) "")))
+         (recipients (rassq-delete-all nil
+                                       (list (cons 'To (alist-get 'to hdrs))
+                                             (cons 'Cc (alist-get 'cc hdrs)))))
+         (to-first (car (bbdb-split "," (or (alist-get 'To recipients) "")))))
+    (dolist (r recipients)
+      (setcdr r (rfc2047-decode-address-string (cdr r))))
+    ;; This is a new message, can directly add to list... but better be safe.
+    (gnus-recent-add-to-list
+     (list (format "-> %s: %s \t%s"
+                   (propertize (gnus-recent-get-email-name to-first t) 'face 'bold)
+                   (alist-get 'subject hdrs)
+                   (propertize date 'face 'gnus-recent-date-face))
+           (cons 'group (alist-get 'gcc hdrs))
+           (cons 'message-id (alist-get 'message-id hdrs))
+           (cons 'date  date)
+           (cons 'subject (alist-get 'subject  hdrs))
+           (cons 'sender author)
+           (cons 'recipients recipients)
+           (cons 'references (alist-get 'references hdrs))
+           (cons 'in-reply-to (alist-get 'in-reply-to hdrs))))))
+
+(defun gnus-recent--get-message-data ()
+  "Get the headers from a new message.
+Returns a header alist, see function
+`mail-header-extract-no-properties'. Needs to run with the
+`message-header-hook' which applies narrowing to the message
+headers by default. Saves the header data to variable
+`gnus-recent--temp-message-header' so it can get out of the way
+as quickly as possible. After the message is sent,
+`gnus-recent--track-message' processes the header data and adds
+an entry to `gnus-recent--articles-list'."
+  (save-restriction
+    (goto-char (point-min))
+    (setq gnus-recent--temp-message-headers (mail-header-extract-no-properties))))
+
+;;
+;; starting gnus-recent
+;;
+(defun gnus-recent-start ()
+  "Start Gnus Recent."
+  (interactive)
+  (gnus-message 5 "Starting gnus-recent")
```

Comment:
` 5 - messages for things
that take a long time, 7 - not very important messages on stuff`

This is probably level 7 (same for a lot of the other gnus-messages here)

Reply:
6 is the default gnus-verbose value., despite what the manual says. So anything from 6 and above will not display.
At this stage and development, it is probably a  good practice to keep a closer tab on the workings than normal. Can always change it as things mature.